lnwallet/channel_test.go

```diff
@@ -7574,8 +7580,17 @@ func TestChannelMaxFeeRate(t *testing.T) {
 	// to slightly lower fee rates.
 	assertMaxFeeRate(anchorChannel, 1.0, 435941555)
 	assertMaxFeeRate(anchorChannel, 0.001, 435941)
-	assertMaxFeeRate(anchorChannel, 0.000001, 435)
-	assertMaxFeeRate(anchorChannel, 0.0000001, chainfee.FeePerKwFloor)
+
+	// When due to the fee allocation the resulting max fee rate would fall
+	// below the current commitment fee rate, we cap it at the current fee
+	// rate which is 6000 sat/kw and was definied when creating the test
+	// channel.
+	// When calculating the max fee of the commitment transaction there
+	// are some minor rounding differences (5999 sat/kw) because when
+	// calculating the actual fee of the commmitment tx the fee is always
+	// rounded down according to BOLT 03.
+	assertMaxFeeRate(anchorChannel, 0.000001, 5999)
```

Comment:
Can we restructure this test by getting it to reveal it's current fee allocation and comparing against that? This number is even more magic than the last one and while the comments indicate to me that this number, magic as it is, is intentional, it doesn't actually do much to persuade me that the property being checked (fees always ratchet upwards) is actually holding.